iron-menu-behavior.html

```diff
@@ -148,7 +148,12 @@
         var item = this.items[(curFocusIndex - i + length) % length];
         if (!item.hasAttribute('disabled')) {
           this._setFocusedItem(item);
-          return;
+
+          // Focus might not have worked, if the element was hidden or not
+          // focusable. In that case, try again.
+          if (document.activeElement == item) {
```

Comment:
sorry, this maybe shows what I'm talking about more clearly: https://jsfiddle.net/umbrstdw/

`document.activeElement` is the container if either children are focused; it's impossible to tell the difference without reaching into the shadow root (accessing `container.shadowRoot.activeElement`).  [`IronOverlayManager#deepActiveElement`](https://github.com/PolymerElements/iron-overlay-behavior/blob/master/iron-overlay-manager.html#L68) handles this, for instance.

if a `<paper-menu>` (or anything that uses `IronMenuBehavior`) is created within a shadow root, checking whether `item == document.activeElement` probably isn't sufficient.

anyways, as long as `[disabled]`, `[hidden]`, `display: none;`, `visibility: hidden;`, and `tabIndex < 0` are accounted for, any solution is fine with me. https://jsfiddle.net/18wfco50/1/